bin/alluxio-start.sh

```diff
@@ -237,6 +238,60 @@ restart_workers() {
   ${LAUNCHER} "${BIN}/alluxio-workers.sh" "${BIN}/alluxio-start.sh" "restart_worker"
 }
 
+get_offline_worker() {
+  local run=
+  local result=""
+  run=$(ps -ef | grep "alluxio.worker.AlluxioWorker" | grep "java" | wc | awk '{ print $1; }')
+  if [[ ${run} -eq 0 ]]; then
+    result=$(hostname -f)
+  fi
+  echo "${result}"
+}
+
+get_offline_workers() {
+  local result=""
+  local run=
+  local i=0
+  local workers=$(cat "${ALLUXIO_CONF_DIR}/workers" | sed  "s/#.*$//;/^$/d")
+  for worker in $(echo ${workers}); do
+    if [[ ${i} -gt 0 ]]; then
+      result+=","
+    fi
+    run=$(ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -tt ${worker} \
+        ps -ef | grep "alluxio.worker.AlluxioWorker" | grep "java" | wc | awk '{ print $1; }')
+    if [[ ${run} -eq 0 ]]; then
+      result+="${worker}"
+    fi
+    i=$((i+1))
+  done
+  echo "${result}"
+}
+
+start_monitor() {
+  local action=$1
+  local nodes=$2
+  local run=
+  if [[ "${action}" == "restart_worker" ]]; then
+    action="worker"
+    if [[ -z "${nodes}" ]]; then
+      run="false"
+    fi
+  elif [[ "${action}" == "restart_workers" ]]; then
+    action="workers"
+    if [[ -z "${nodes}" ]]; then
+      run="false"
+    fi
+  elif [[ "${action}" == "logserver" || "${action}" == "run_safe" ]]; then
```

Comment:
should this just be "safe"? perhaps we should define the modes with constants to mitigate issues like this

Reply:
Yes, it's not nice... I guess this is the first time we need to filter by action, perhaps I could pass an additional parameter instead?

Reply:
Done

Reply:
I mean there is no action called "run_safe", but there is an action called "safe"

Reply:
@aaudiber this was fix in the last commit.

Reply:
sorry, must have been looking at outdated code